target_chains/solana/programs/oracle-instance/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+use {
+    anchor_lang::prelude::*,
+    pyth_solana_receiver::{
+        cpi::accounts::PostUpdate,
+        program::PythSolanaReceiver,
+        PostUpdateParams,
+    },
+    pyth_solana_receiver_sdk::{
+        config::{
+            Config,
+            DataSource,
+        },
+        price_update::{
+            PriceUpdateV1,
+            VerificationLevel,
+        },
+    },
+    pythnet_sdk::{
+        accumulators::merkle::MerkleRoot,
+        hashers::keccak256_160::Keccak160,
+        messages::{
+            FeedId,
+            Message,
+        },
+        wire::{
+            from_slice,
+            v1::{
+                MerklePriceUpdate,
+                WormholeMessage,
+                WormholePayload,
+            },
+        },
+    },
+    solana_program::{
+        keccak,
+        program_memory::sol_memcpy,
+        secp256k1_recover::secp256k1_recover,
+        system_instruction,
+    },
+    wormhole_core_bridge_solana::{
+        sdk::{
+            legacy::AccountVariant,
+            VaaAccount,
+        },
+        state::GuardianSet,
+    },
+    wormhole_raw_vaas::{
+        utils::quorum,
+        GuardianSetSig,
+        Vaa,
+    },
+};
+
+// pub mod error;
+pub mod sdk;
+
+declare_id!("F9SP6tBXw9Af7BYauo7Y2R5Es2mpv8FP5aNCXMihp6Za");
+
+#[error_code]
+pub enum OracleInstanceError {
+    #[msg("Updates must be monotonically increasing")]
+    UpdatesNotMonotonic,
+    #[msg("Price feed id mismatch")]
+    PriceFeedMessageMismatch
+}
+#[program]
+pub mod oracle_instance {
+
+    use super::*;
+
+    pub fn update_price_feed(
+        ctx: Context<UpdatePriceFeed>,
+        params: PostUpdateParams,
+        feed_id: FeedId,
+    ) -> Result<()> {
+        let cpi_program = ctx.accounts.pyth_solana_receiver.to_account_info().clone();
+        let cpi_accounts = PostUpdate {
+            payer:                ctx.accounts.payer.to_account_info().clone(),
+            write_authority : ctx.accounts.price_feed_account.to_account_info().clone(),
+            encoded_vaa:          ctx.accounts.encoded_vaa.to_account_info().clone(),
+            config:               ctx.accounts.config.to_account_info().clone(),
+            treasury:             ctx.accounts.treasury.to_account_info().clone(),
+            price_update_account: ctx.accounts.price_feed_account.to_account_info().clone(),
+            system_program:       ctx.accounts.system_program.to_account_info().clone(),
+        };
+
+        let seeds = &[
+            feed_id.as_ref(),
```

Comment:
reminder to update the seeds here if you change the seeds per the comment below

Reply:
Done